src/transformers/models/falcon_mamba/modeling_falcon_mamba.py

```diff
@@ -745,11 +773,17 @@ def prepare_inputs_for_generation(
         else:
             model_inputs = {"input_ids": input_ids.contiguous()}
 
+        # In case cache is not used, manually add a new column in the attention mask
+        if not use_cache and attention_mask is not None and input_ids.shape != attention_mask.shape:
+            pad_length = input_ids.shape[-1] - attention_mask.shape[-1]
+            attention_mask = torch.cat([attention_mask, torch.ones_like(input_ids[:, :pad_length])], dim=-1)
```

Comment:
Not sure I understand why we are adding a `[1] x batch_size`? ( past_length is usually gonna be `1 - current_generation_token` , so imagine 20 input ids, then -19 to slice the input_ids? 
Unless the inpud_ids is 20, but then it always has the same shape as the mask

Reply:
This is for users that run generation with `use_cache=False` and makes sure to manually update the attention mask because this is done no where else except here

Reply:
then this is more a problem with `generate` as it should pass the correct attention mask 😓